api/envoy/api/v2/core/health_check.proto

```diff
@@ -267,4 +279,6 @@ message HealthCheck {
   // initial health check failure event will be logged.
   // The default value is false.
   bool always_log_health_check_failures = 19;
+
+  TLSOptions tls_options = 21;
```

Comment:
Needs a field level comment (this is where the generated docs will render the user facing description).